### ISSUE TITLE
Disable "Limit" with Query Parameters

### ISSUE THREAD

**Issue by johannesschobel** [andersao/l5-repository#409]
Hey @andersao and community,

thank you very much for this awesome package.
I have an issue, which I am currently not able to resolve by myself - however, maybe you can help me with this problem!?

Consider the following example:
I have a route, that returns all entities of a respective repository, like so:
```php
return $this->repository->paginate();
```
where `$this->repository` is the respective repository.

As I am using `paginate()`, all results are paginated (e.g., only 20 entries per page are displayed), which is a desired behaviour for my mobile application clients. However, web-clients (which may have a better connectivity) should receive all entities.

Is there a way to "disable" the `limit` query parameter? My current "workaround" is to make a request with `?limit=100000`. Clearly, this is not a good solution - can you provide me with any better idea? Is there a built-in-way to do this?!

Cheers and thanks a lot!

**Comment by johannesschobel** [comment]
anyone has some idea on this one? @eullercdr , @andersao or community?

**Comment by ptsilva** [comment]
Use all() instead of paginate ()

**Comment by johannesschobel** [comment]
yeah, but then I loose the ability to paginate the results.. I would like to disable the pagination, *if the user* decides to.. e.g., if the appends the query parameter `?limit=0` or so...

**Comment by ptsilva** [comment]
Can you use the LengthAwarePaginator.
Boot the paginator using all() result, and then make your control of limit/pages.

https://github.com/laravel/framework/blob/f2e485960b7fe8b0ddbf2fc2e76aff5339948621/src/Illuminate/Pagination/LengthAwarePaginator.php#L41

**Comment by johannesschobel** [comment]
I thought, it would be possible to use this "out of the box"..
Would you accept a PR to fix this behaviour?

**Comment by ptsilva** [comment]
I don't think that this a problem. But, @andersao or @eullercdr maybe accept the PR.

**Comment by duongtdvn** [comment]
Im in the same situation and there is my current walk around:
```php
$results = $this->repository;

return request()->has('limit') ? $results->paginate(request('limit')) : $results->all();
```

for laravel 5.5:
```php
$results = $this->repository;

return request()->filled('limit') ? $results->paginate(request('limit')) : $results->all();
```

notice the `request()->has()` has been replaced with `request()->filled` since it validate that the parameter is not empty as [https://laravel.com/docs/5.5/requests](url)
